client/rest-high-level/src/test/java/org/elasticsearch/client/RequestTests.java

```diff
@@ -1018,6 +1020,56 @@ public void testExistsAliasNoAliasNoIndex() {
         assertEquals("existsAlias requires at least an alias or an index", iae.getMessage());
     }
 
+    public void testSplit() throws IOException {
+        resizeTest(ResizeType.SPLIT, Request::split);
+    }
+
+    public void testSplitWrongResizeType() {
+        ResizeRequest resizeRequest = new ResizeRequest("target", "source");
+        resizeRequest.setResizeType(ResizeType.SHRINK);
+        IllegalArgumentException iae = expectThrows(IllegalArgumentException.class, () -> Request.split(resizeRequest));
+        assertEquals("Wrong resize type for indices.split request: SHRINK", iae.getMessage());
+    }
+
+    public void testShrinkWrongResizeType() {
+        ResizeRequest resizeRequest = new ResizeRequest("target", "source");
+        resizeRequest.setResizeType(ResizeType.SPLIT);
+        IllegalArgumentException iae = expectThrows(IllegalArgumentException.class, () -> Request.shrink(resizeRequest));
+        assertEquals("Wrong resize type for indices.shrink request: SPLIT", iae.getMessage());
+    }
+
+    public void testShrink() throws IOException {
+        resizeTest(ResizeType.SHRINK, Request::shrink);
+    }
+
+    private static void resizeTest(ResizeType resizeType, CheckedFunction<ResizeRequest, Request, IOException> function)
+            throws IOException {
+        ResizeRequest resizeRequest = new ResizeRequest(randomAlphaOfLengthBetween(3, 10), randomAlphaOfLengthBetween(3, 10));
+        resizeRequest.setResizeType(resizeType);
+        Map<String, String> expectedParams = new HashMap<>();
+        setRandomWaitForActiveShards(resizeRequest::setWaitForActiveShards, expectedParams);
+        setRandomMasterTimeout(resizeRequest, expectedParams);
+        setRandomTimeout(resizeRequest::timeout, resizeRequest.timeout(), expectedParams);
+
+        if (randomBoolean()) {
+            CreateIndexRequest createIndexRequest = new CreateIndexRequest(randomAlphaOfLengthBetween(3, 10));
+            if (randomBoolean()) {
+                createIndexRequest.settings(randomIndexSettings());
+            }
+            if (randomBoolean()) {
+                randomAliases(createIndexRequest);
+            }
+        }
```

Comment:
no, resize supports only a couple of sections compared to create index (aliases and settings only). I think it is better to set only those specifically.

Reply:
My bad ! 
But I do not see it being used here. Did you mean to set it as target?

Reply:
it should be set within the method, that is why we pass in the request, otherwise we'd have to return a `List<Alias>` and set them one by one as the request doesn't accept a list of alias directly.

Reply:
ops. thanks for being persistent on this :)